src/main/java/redis/clients/jedis/DefaultJedisClientConfig.java

```diff
@@ -65,6 +66,21 @@ public String getPassword() {
     return password;
   }
 
+  /**
+   * @param user has to be the same one with which the factory is created
+   * @param password
+   * @throws IllegalArgumentException if the user is not the original user
+   */
+  @Override
+  public synchronized void updatePassword(String user, String password) {
+    if (!Objects.equals(this.user, user)) {
+      throw new IllegalArgumentException();
+    }
```

Comment:
There's a little confusion here. Will there can pass different users here ? Currently, we don't event provide setter method for user.

Reply:
> Will there can pass different users here ?

No.

I thought it'd be better to have some soft of _safety check_ to fail before waiting to fail during AUTH command. But both @gkorland and you got confused here. May be, this is not the way to do it.

Reply:
Remove checking of user make sense to me.